packages/utils/src/commonwell/cq-org-builder.ts

```diff
@@ -0,0 +1,176 @@
+  // so it does allow you to search through all Surescripts orgs.
+  // https://metriport.slack.com/archives/C04DMKE9DME/p1700501978325669?thread_ts=1700077189.772289&cid=C04DMKE9DME
+  "Surescripts",
+];
+
+type OrgWithGateway = CQOrg & { gateway: string };
+
+type Params = {
+  local?: boolean;
+};
+const program = new Command();
+program
+  .name("cq-org-builder")
+  .description("CLI to rebuild the CW's CQ org list.")
+  .option(
+    `--local`,
+    "Rebuild the lisf from the local file without calling the Sequoia API (does not update the list of US states"
+  )
+  .showHelpAfterError();
+
+export async function main() {
+  program.parse();
+  const { local: isLocal } = program.opts<Params>();
+
+  console.log(`Runnning at ${new Date().toISOString()} - local: ${isLocal}`);
+
+  fs.writeFileSync(outputFailedFilename, "");
+
+  const gateways = originalPayloadFromCW as Gateway[];
+  console.log(`I have ${gateways.length} gateways on the original list...`);
+  const gatewaysToProcess = gateways.filter(g => !excludeGatewayNames.includes(g.Name));
+  console.log(
+    `...processing ${gatewaysToProcess.length} gateways (exclude list has ${excludeGatewayNames.length} items)`
+  );
+
+  const totalFailed: { gateway: string; reason: string }[] = [];
+  const flatOrgs: CQOrgHydrated[] = [];
+  for (const gateway of gatewaysToProcess) {
+    const orgs: OrgWithGateway[] = gateway.Organizations.map(o => {
+      return { id: o.Id, name: o.Name, gateway: gateway.Name };
+    });
+    const res = await executeAsynchronously(
+      orgs,
+      async org => {
+        const states: string[] = isLocal
+          ? await getStatesFromCurrent(org.id)
+          : await getOrgStatesFromSequoia(org.id);
+        flatOrgs.push({ ...org, states: states, prio: getPrio(org) });
+      },
+      { numberOfParallelExecutions: 100, keepExecutingOnError: true }
+    );
+    const failed = res.flatMap(r => (r.status === "rejected" ? r.reason : []));
+    const success = res.flatMap(r => (r.status === "fulfilled" ? "ok" : []));
+    console.log(
+      `--------------> Done w/ orgs of GW ${gateway.Name}: ${success.length} succeeded, ${failed.length} failed`
+    );
+    if (failed.length) {
+      fs.appendFileSync(outputFailedFilename, JSON.stringify(failed, null, 2));
+      const errorsOnThisGW = uniqBy(failed, f => f.reason);
+      totalFailed.push(...errorsOnThisGW.map(f => ({ gateway: gateway.Name, reason: f.reason })));
+    }
+  }
+  fs.writeFileSync(resultFilename, JSON.stringify(flatOrgs, null, 2));
+
+  console.log(`Saved flat org list to ${resultFilename}`);
+
+  console.log(`Failed`, totalFailed);
+
+  console.log(`Done at ${new Date().toISOString()}`);
+}
+
+async function getInfo(url: string) {
+  return retryFunction(() => axios.get(url), 5, 500);
+}
+
+async function getOrgStatesFromSequoia(orgOID: string): Promise<string[]> {
+  if (!orgOID) return [];
+  const url =
```

Comment:
nit: use URLSearchParams instead? :P

Reply:
yeah, I just copy/paste this code and did some minor adjustments, but I agree. Updating 👍